hw/ip/entropy_src/doc/_index.md

```diff
@@ -0,0 +1,210 @@
+The seed register can be written without restriction, regardless of any state of any control bits.
+Once the `es_enable` bit is set, subsequent loads of the seed into the LFSR will occur only when only the `es_init` bit is set, and the state of the FIFO is full.
+The intent for this is that the FIFO is in a safe state, and any updates from the LFSR will be ignored.
+
+Another control bit called `lock_seed` will prevent any firmware sequence from updating the seed value into the LFSR, even though the seed register itself may still be altered.
+Only a system reset will undo the `lock_seed` state of operation.
+
+The seed register can be updated at any time while the Noise FIFO is in operational mode.
+As long as the update sequence is followed by firmware, there will be no timing hazards when accessing these registers by the software.
+
+### Noise Processing
+
+When enabled, the ENTROPY_SRC block will generate noise bits continuously.
+The `Entropy ready` bit in the {{< regref "ESSTATUS" >}} register will signal to the software when noise bits should read from the {{< regref "ESNOISE" >}} register.
+While the software is reading the FIFO, additional noise bits are being captured by the FIFO as long as the FIFO is not full.
+The software will do 32-bit register reads of the {{< regref "ESNOISE" >}} register to retrieve the noise bits.
+Each read will automatically pop an entry off the head of the FIFO.
+New noise bits can enter the FIFO at the tail concurrently.
+
+Since the noise flow is continuous, these is no tracking state machine required for the design.
+Instead, the threshold register is used to set the point of notification to the software.
+Notification can be either by polling or by an interrupt.
+For debug, a FIFO depth register can be read to find out the current state of the FIFO at any time. 
+
+An additional feature is the presence of a noise rate register.
+The purpose of this register is to simulation slower noise generation sources.
+This will aid firmware design and debug when trying to build handle all noise source types and rates. 
+
+### Interrupts
+
+The ENTROPY_SRC module has two interrupts: `es_noise_valid` and `es_noise_fifo_err`.
+
+The `es_noise_valid` interrupt should be used when a noise source has been implemented that is relatively slow.
+For this revision, the noise source is very fast, and polling the es_status bit should be done.
+
+The `es_noise_fifo_err` interrupt will fire when the noise FIFO has a malfunction.
+The conditions that cause this to happen are either when there is a push to a full FIFO or a pull from an empty FIFO.
+
+
+### Noise FIFO Operational Sequence
+
+{{< wavejson >}}
+{signal: [
+   {name: 'clk'                 , wave: 'p.......|............'},
+   {name: 'es_enable'           , wave: '01......|............'},
+   {name: 'es_init'             , wave: '010.....|............'},
+   {name: 'fifo_push_capt_noise', wave: '0..1...0|.10.10.10.10', data: ['ph0','ph1','ph2','ph3']},
```

Comment:
another WF bug? You have `fifo_push_capt_noise` true for 4 cycles, but only three `ES` values in `noise_bits`. This could be a bug in `wavedrom` or on my side only? In my view the `6` values are completely dropped, rather than rendering as a 4th color. The problem appears to only be for `6` and higher. If you're seeing the same thing, then we can debug. Otherwise might be just something on my side. It appears to work fine for wavedrom tutorials.
 
![Screenshot from 2020-01-02 18-07-11](https://user-images.githubusercontent.com/41092437/71703664-bbef9980-2d8a-11ea-9844-1b2b8b371204.png)